github-actions/unified-status-check/src/pull-request.ts

```diff
@@ -5,59 +5,146 @@ import {CheckConclusionState, PullRequestState, StatusState} from '@octokit/grap
+  state = this.pullRequest.state;
+  /** The list of current labels applied to the pull request. */
+  labels = this.pullRequest.labels.nodes.map(({name}) => name);
+  /** The statuses for the pull request. */
+  statuses = {
+    /** All statuses currently in a pending state. */
+    pending: [] as NormalizedStatus[],
+    /** All statuses currently in a success state. */
+    success: [] as NormalizedStatus[],
+    /** All statuses currently in a failing state. */
+    failure: [] as NormalizedStatus[],
+    /** All  non-ignored statuses on the pull request. */
+    all: [] as NormalizedStatus[],
+    /** All the statuses which have been ignored. */
+    ignored: [] as NormalizedStatus[],
+  };
+  /** The id of the previous checkRun to be updated, if it exists. */
+  previousRunId: number | undefined;
 
-class Statuses {
-  pending: NormalizedStatus[] = [];
-  failing: NormalizedStatus[] = [];
-  passing: NormalizedStatus[] = [];
-  ignored: NormalizedStatus[] = [];
-  all: NormalizedStatus[] = [];
-  unifiedCheckStatus: NormalizedStatus | undefined;
+  /** Retrieve the pull request from Github. */
+  static async get(github: Octokit) {
+    const pullRequest = await github.graphql<typeof PR_SCHEMA>(query(PR_SCHEMA).toString());
+    return new PullRequest(github, pullRequest.repository.pullRequest);
+  }
 
-  populate(statuses: NormalizedStatus[]) {
-    const ignored = [unifiedStatusCheckName, ...ignoredStatuses];
+  private constructor(private github: Octokit, private pullRequest: PullRequestFromGithub) {
+    /** The list of all checks and statuses on the pull request, if any exist. */
+    const checksAndStatuses = pullRequest.commits.nodes[0].commit.statusCheckRollup?.contexts.nodes;
+    if (checksAndStatuses) {
+      this.normalizeAndPopulateStatuses(checksAndStatuses);
+    }
+  }
 
-    for (const status of statuses) {
-      if (ignored.some((matcher) => status.name.match(matcher))) {
-        this.ignored.push(status);
-        if (status.name === unifiedStatusCheckName) {
-          this.unifiedCheckStatus = status;
+  /** Normalize the statuses and checks for usage on the pull request, and then populate the statuses values. */
+  private normalizeAndPopulateStatuses(checksAndStatuses: CheckOrStatus[] = []) {
+    checksAndStatuses.forEach((checkOrStatus: CheckOrStatus) => {
+      /** The normalized status. */
+      let status: NormalizedStatus;
+
+      if (checkOrStatus.__typename === 'CheckRun') {
+        status = {
+          state: checkOrStatus.conclusion
+            ? checkConclusionStateToStatusStateMap.get(checkOrStatus.conclusion)!
+            : 'pending',
+          description: checkOrStatus.title || undefined,
+          name: checkOrStatus.name || 'unknown-check-run',
+        };
+
+        // If the check is the previous result for the unified check, store the previous run id.
+        if (checkOrStatus.name === PullRequest.checkName) {
+          this.previousRunId = checkOrStatus.databaseId;
```

Comment:
is this really called `databaseId`? just asking out of curiosity. seems odd from GitHub

Reply:
Yeah, it seems to be.  I was surprised as well.